src/main/scala/rules/Evaluator.scala

```diff
@@ -918,7 +918,8 @@ object evaluator extends EvaluationRules {
                                  * incomplete).
                                  */
                              smDomainNeeded = true,
-                             moreJoins = JoinMode.Off)
+                             moreJoins = JoinMode.Off,
+                             assertReadAccessOnly = if (Verifier.config.respectFunctionPrePermAmounts()) s2.assertReadAccessOnly else true)
```

Comment:
why don't we have `false` here when `Verifier.config.respectFunctionPrePermAmounts()` holds?

Reply:
Ah, I think this is actually correct. Here we create the state in which the function precondition is consumed, so if we're in legacy mode, then the flag doesn't change (presumably it stays false), but otherwise we (temporarily) set it to true s.t. permission amounts are ignored during this exhale.